### ISSUE TITLE
Use CMake djinni-support-lib target in integration tests

### ISSUE THREAD

**Issue by jothepro** [cross-language-cpp/djinni-support-lib#22]
Currently the integration tests do include the support-lib [by globbing for it's source files](https://github.com/cross-language-cpp/djinni-support-lib/blob/main/test-suite/CMakeLists.txt#L106) like in the good old days:

```cmake
if(DJINNI_WITH_JNI)
  file(GLOB_RECURSE SUPPORT_LIB_JNI_SRCS "${CMAKE_SOURCE_DIR}/djinni/jni/*")

  add_library(DjinniTestNative SHARED
    ${SUPPORT_LIB_JNI_SRCS}
    ${JNI_GENERATED_SRCS}
    ${CPP_GENERATED_SRCS}
    ${WCHAR_JNI_GENERATED_SRCS}
    ${WCHAR_CPP_GENERATED_SRCS}
    ${CPP_HANDWRITTEN_SRCS}
  )
```

This is not optimal as the tests can not detect errors in the support-lib CMake target configuration.

As a maintainer I would like the tests to link the [djinni-support-lib target](https://github.com/cross-language-cpp/djinni-support-lib/blob/main/CMakeLists.txt#L32) that is defined in the root CMakeLists.txt:

```cmake
if(DJINNI_WITH_JNI)
  add_library(DjinniTestNative SHARED
    ${JNI_GENERATED_SRCS}
    ${CPP_GENERATED_SRCS}
    ${WCHAR_JNI_GENERATED_SRCS}
    ${WCHAR_CPP_GENERATED_SRCS}
    ${CPP_HANDWRITTEN_SRCS}
  )
  target_link_libraries(DjinniTestNative PUBLIC djinni_support_lib)
```

**Comment by a4z** [comment]
Good catch! This is definitely something we have to do, and from that POV its more than an improvement, its a requirement.

**Comment by jothepro** [comment]
This has been fixed with #26
